lib/sshkit/backends/abstract.rb

```diff
@@ -82,9 +82,10 @@ def execute(*args)
 
       def within(directory, &_block)
         (@pwd ||= []).push directory.to_s
+        escaped = pwd_path.shellescape.gsub("\\~", "~")
```

Comment:
What do you think about extracting this double-escape logic into a separate utility method for clarity? The method could be called e.g. `shellescape_except_tilde` with a documentation comment explaining why it is desirable. Then we could reuse that method here and in `command.rb`. I am not sure where such a method would go since we don't really have a "utils" module; perhaps it could just be on the `SSHKit` module itself, for lack of a more obvious location.